devlab/tests/base.py

```diff
@@ -274,14 +274,28 @@ def check_snapshot_state(self, snp_id):
         return False
 
     @staticmethod
+    def check_floating_ip_assigned(vm, floating_ip_address):
+        for net in vm.addresses:
+            for addr in vm.addresses[net]:
+                if addr['OS-EXT-IPS:type'] == 'floating':
+                    return True
+        try:
+            vm.add_floating_ip(floating_ip_address)
+        except ks_exceptions.BadRequest:
+            pass
+        return False
+
+    @staticmethod
     def wait_until_objects_created(obj_list, check_func, timeout):
         obj_list = obj_list[:]
         waiting = 0
         delay = 1
         while waiting < timeout:
             for obj in obj_list[:]:
-                if check_func(obj):
-                    obj_list.remove(obj)
+                if not isinstance(obj, tuple):
+                    obj = (obj, )
+                if check_func(*obj):
+                    obj_list.remove(obj[0])
```

Comment:
Looks like the `remove` will not remove object from the list if it's a tuple of more than 1 element